types/null/uint8.go

```diff
@@ -86,51 +86,97 @@ func (i Uint8) Value() (driver.Value, error) {
 
 // Scan implements the database/sql Scanner interface. It will receive a value
 // from an SQL database and assign it to i, so long as the provided data is of
-// type nil, uint8, string, or another integer type that doesn't overflow uint8.
-// All other types will result in an error.
+// type nil, uint8, string, or another integer or float type that doesn't
+// overflow uint8. All other types will result in an error.
 func (i *Uint8) Scan(src interface{}) error {
+	// Helpers for repetitive int scans
+	scanUint := func(src interface{}, vi uint64) error {
+		if vi > math.MaxUint8 {
+			return fmt.Errorf("null.Uint8: failed to scan type %T (%v): overflow", src, src)
+		}
+		i.Uint8 = uint8(vi)
+		i.Valid = true
+		return nil
+	}
+	scanInt := func(src interface{}, vi int64) error {
+		if vi > math.MaxUint8 {
+			return fmt.Errorf("null.Uint8: failed to scan type %T (%v): overflow", src, src)
+		} else if vi < 0 {
+			return fmt.Errorf("null.Uint8: failed to scan type %T (%v): negative value", src, src)
+		}
+		i.Uint8 = uint8(vi)
+		i.Valid = true
+		return nil
+	}
+
+	if src == nil {
+		i.Uint8 = 0
+		i.Valid = false
+		return nil
+	}
 	if i == nil {
 		return fmt.Errorf("null.Uint8: Scan called on nil pointer")
 	}
+
 	switch val := src.(type) {
 	case uint8:
 		i.Uint8 = val
 		i.Valid = true
 		return nil
+	case uint:
+		return scanUint(src, uint64(val))
+	case uint16:
+		return scanUint(src, uint64(val))
+	case uint32:
+		return scanUint(src, uint64(val))
+	case uint64:
+		return scanUint(src, uint64(val))
```

Comment:
why aren't these `case` statements grouped? `case uint, uint16, uint32, uint64:`?

Reply:
The compiler complains that the type hasn't been properly asserted. It would certainly be nicer if I could group them like that.